proto/backend.proto

```diff
@@ -306,3 +306,160 @@ message StreamPacket {
+}
+
+message InstanceSettingsAdmissionRequest {
+  // NOTE: this may not include app or datasource instance settings depending on the request
+  PluginContext pluginContext = 1;
+
+  // The requested operation
+  StorageOperation operation = 2;
+
+  // Requested app instance state (not yet saved)
+  AppInstanceSettings appInstanceSettings = 3;
+
+  // Requested data source instance state (not yet saved)
+  DataSourceInstanceSettings dataSourceInstanceSettings = 4;
+}
+
+message InstanceSettingsResponse {
+  // Allowed indicates whether or not the admission request was permitted.
+  bool allowed = 1;
+
+  // Result contains extra details into why an admission request was denied.
+  // This field IS NOT consulted in any way if "Allowed" is "true".
+  // +optional
+  StatusResult result = 2;
+
+  // warnings is a list of warning messages to return to the requesting API client.
+  // Warning messages describe a problem the client making the API request should correct or be aware of.
+  // Limit warnings to 120 characters if possible.
+  // Warnings over 256 characters and large numbers of warnings may be truncated.
+  // +optional
+  repeated string warnings = 3;
+
+  // Valid app instance settings if they exist in the request
+  AppInstanceSettings appInstanceSettings = 4;
+
+  // Valid datasource instance settings if they exist in the request
+  DataSourceInstanceSettings dataSourceInstanceSettings = 5;
+}
+
+// AdmissionRequest contains information from a kubernetes Admission request and decoded object(s).
+// See: https://github.com/kubernetes/kubernetes/blob/v1.30.0/pkg/apis/admission/types.go#L41
+// And: https://github.com/grafana/grafana-app-sdk/blob/main/resource/admission.go#L14
+// NOTE: this does not include a plugin context
+message AdmissionRequest {
+  // NOTE: this may not include app or datasource instance settings depending on the request
+  PluginContext pluginContext = 1;
+
+  // The requested operation
+  StorageOperation operation = 2;
+  
+  // Object is the object in the request.  This includes the full metadata envelope.
+  // The Group+Version+Kind will be included in the payload
+  bytes object_bytes = 3;
+
+  // OldObject is the object as it currently exists in storage. This includes the full metadata envelope.
+  bytes old_object_bytes = 4;
+}
+
+// ConversionRequest supports converting an object from on version to another 
+message ConversionRequest {
+  // NOTE: this context exists because most of the middleware depends on it
+  // The format conversions most likely do not depend on instance settings passed with the context
+  PluginContext pluginContext = 1;
+
+  enum ObjectEnvelope {
+    // Kubernetes style resource
+    RESOURCE = 0;
+    // Full query request
+    QUERY_DATA_REQUEST = 1;
+  }
+
+  // object metadata wrapper
+  ObjectEnvelope envelope = 2;
+  
+  // Object is the object in the request.  This includes the full metadata envelope.
+  bytes object_bytes = 3;
```

Comment:
The version for the object is part of these bytes, right? I'm pretty sure the answer is "yes"

Reply:
yes in k8s land, this is the full object with metadata